UI/undo-stack-obs.cpp

```diff
@@ -19,6 +21,13 @@ void undo_stack::add_action(const QString &name, undo_redo_cb undo,
 			    undo_redo_cb redo, std::string undo_data,
 			    std::string redo_data, func d)
 {
+	if (undo_items.size() >= MAX_STACK_SIZE) {
```

Comment:
Should this not be a while instead of an if? That way if it ever ends up over 5000, it'll loop till it's below 5000?